### PR TITLE
Add output variable support for container less plugins

### PR DESCRIPTION
**Pull request by raghavharness**

### CODE REVIEW

File: pipeline/runtime/common.go [hunk 1]
```diff
@@ -70,6 +74,10 @@ func isPython(entrypoint []string) bool {
 	return false
 }
 
+func isContainerlessPlugin(image string, entrypoint []string) bool {
+	return image == "" && len(entrypoint) > 0 && entrypoint[0] == "plugin"
```
Comment (shubham149) [inline pipeline/runtime/common.go:78]: this does not seem right. lite-engine should not be aware of plugin binary

File: pipeline/runtime/common.go [hunk 2]
```diff
@@ -118,6 +118,25 @@ func fetchExportedEnvVars(envFile string, out io.Writer) map[string]string {
 	return env
 }
 
+// Fetches output variables exported by the step.
+func fetchOutputVariablesFromEnvFile(outputFile string, out io.Writer) map[string]string {
```
Comment (shubham149) [inline pipeline/runtime/common.go:122]: this method is a copy of above one. Can we de-duplicate it?

File: pipeline/runtime/run.go [hunk 3]
```diff
@@ -34,7 +34,9 @@ func executeRunStep(ctx context.Context, engine *engine.Engine, r *api.StartStep
 		return nil, nil, nil, fmt.Errorf("output variable should not be set for unset entrypoint or command")
 	}
 
-	outputFile := fmt.Sprintf("%s/%s.out", pipeline.SharedVolPath, step.ID)
+	outputFile := fmt.Sprintf("%s/%s.env", pipeline.SharedVolPath, step.ID)
```
Comment (shubham149) [inline pipeline/runtime/run.go:37]: %s/%s-output.env

Comment (shubham149) [reply pipeline/runtime/run.go:37]: So, that confusion is not present with DRONE_ENV one

File: pipeline/runtime/common.go (hunk 2 above)
Comment (raghavharness) [reply pipeline/runtime/common.go:122]: Let me make a common one.

File: pipeline/runtime/run.go (hunk 3 above)
Comment (raghavharness) [reply pipeline/runtime/run.go:37]: Sure

File: pipeline/runtime/common.go [hunk 4]
```diff
@@ -101,19 +101,21 @@ func fetchOutputVariables(outputFile string, out io.Writer) (map[string]string,
...
+// Fetches variable in env file exported by the step.
+func fetchExportedVarsFromEnvFile(envFile string, out io.Writer) map[string]string {
 	log := logrus.New()
 	log.Out = out
 
+	env := make(map[string]string)
+
 	if _, err := os.Stat(envFile); errors.Is(err, os.ErrNotExist) {
-		return nil
+		return env
```
Comment (shubham149) [inline pipeline/runtime/common.go:112]: we can return nil here

Comment (shubham149) [reply pipeline/runtime/common.go:112]: Remove fetchOutputVariables method

Comment (raghavharness) [reply pipeline/runtime/common.go:79]: okay

**Review by shubham149 — APPROVED** [review]

**Pull request merged**